packages/plugin-session-replay-browser/src/session-replay.ts

```diff
@@ -76,39 +76,46 @@ class SessionReplay implements SessionReplayEnrichmentPlugin {
       };
     }
 
-    const GlobalScope = getGlobalScope();
-    if (GlobalScope && GlobalScope.window) {
-      GlobalScope.window.addEventListener('blur', () => {
-        this.stopRecordingEvents && this.stopRecordingEvents();
-        this.stopRecordingEvents = null;
-      });
-      GlobalScope.window.addEventListener('focus', () => {
-        void this.initialize();
-      });
+    const globalScope = getGlobalScope();
+    if (globalScope) {
+      globalScope.addEventListener('blur', this.blurListener);
+      globalScope.addEventListener('focus', this.focusListener);
     }
 
-    if (GlobalScope && GlobalScope.document && GlobalScope.document.hasFocus()) {
+    if (globalScope && globalScope.document && globalScope.document.hasFocus()) {
       await this.initialize(true);
     }
   }
 
+  blurListener = () => {
+    this.stopRecordingAndSendEvents();
+  };
+  focusListener = () => {
+    void this.initialize();
+  };
+
+  stopRecordingAndSendEvents(sessionId?: number) {
+    this.stopRecordingEvents && this.stopRecordingEvents();
+    this.stopRecordingEvents = null;
+    const sessionIdToSend = sessionId || this.config.sessionId;
```

Comment:
when would this be different?

Reply:
It's a safety check for the end session event, I want to make sure we send events with the session id associated with the end session event, not the next session id that gets created when a start session event fires